discussions/api_test.py

```diff
@@ -138,20 +149,158 @@ def test_update_discussion_user_no_update(mock_staff_client):
     assert mock_staff_client.users.update.call_count == 0
 
 
-@pytest.mark.parametrize('status_code', [201, 300, 301, 400, 401, 403, 500])
-def test_update_discussion_user_error(mock_staff_client, status_code):
-    """Verify update_discussion_user handles non-200 status codes"""
-    mock_staff_client.users.update.return_value.status_code = status_code
+def test_update_discussion_user_error(mock_staff_client):
+    """Verify update_discussion_user handles non-2xx status codes"""
+    mock_staff_client.users.update.return_value.raise_for_status.side_effect = HTTPError
     with mute_signals(post_save):
         profile = ProfileFactory.create()
     discussion_user = DiscussionUser.objects.create(user=profile.user, username='username')
     with pytest.raises(DiscussionUserSyncException) as exc:
         api.update_discussion_user(discussion_user)
 
-    assert str(exc.value) == "Error updating discussion user, got status_code {}".format(status_code)
+    assert str(exc.value) == "Error updating discussion user for {}".format(profile.user.username)
+
+
+def test_add_to_channel(mock_staff_client):
+    """add_to_channel should add user as contributor and subscriber"""
+    channel_name = 'channel'
+    discussion_username = 'username'
+    api.add_to_channel(channel_name, discussion_username)
+    mock_staff_client.channels.add_contributor.assert_called_once_with(channel_name, discussion_username)
+    mock_staff_client.channels.add_subscriber.assert_called_once_with(channel_name, discussion_username)
+
+
+def test_add_to_channel_failed_contributor(mock_staff_client):
+    """add_to_channel should raise an exception if it fails to add a contributor"""
+    mock_staff_client.channels.add_contributor.return_value.raise_for_status.side_effect = HTTPError
+    with pytest.raises(ContributorSyncException) as ex:
+        api.add_to_channel('channel', 'user')
+    assert ex.value.args[0] == 'Error adding contributor user to channel channel'
+    assert mock_staff_client.channels.add_subscriber.called is False
+
+
+def test_add_to_channel_failed_subscriber(mock_staff_client):
+    """add_to_channel should raise an exception if it fails to add a subscriber"""
+    channel_name = 'channel'
+    discussion_username = 'username'
+    mock_staff_client.channels.add_subscriber.return_value.raise_for_status.side_effect = HTTPError
+    with pytest.raises(SubscriberSyncException) as ex:
+        api.add_to_channel(channel_name, discussion_username)
+    assert ex.value.args[0] == 'Error adding subscriber {user} to channel {channel}'.format(
+        user=discussion_username,
+        channel=channel_name,
+    )
+
+    mock_staff_client.channels.add_contributor.assert_called_once_with(channel_name, discussion_username)
+    mock_staff_client.channels.add_subscriber.assert_called_once_with(channel_name, discussion_username)
+
+
+@pytest.mark.parametrize("contributor_status_code,subscriber_status_code", [
+    (200, 200),
+    (404, 404),
+    (409, 404),
```

Comment:
can we use `rest_framework.status` instead of pure numbers? (same for all the other tests)